src/plasma/plasma_client.cc

```diff
@@ -465,13 +469,67 @@ void plasma_contains(PlasmaConnection *conn, ObjectID obj_id, int *has_object) {
   }
 }
 
+void compute_block_hash(const unsigned char *data, int64_t nbytes, uint64_t *hash) {
+  XXH64_state_t hash_state;
+  XXH64_reset(&hash_state, XXH64_DEFAULT_SEED);
+  XXH64_update(&hash_state, data, nbytes);
+  *hash = XXH64_digest(&hash_state);
+}
+
+bool compute_object_hash_parallel(XXH64_state_t *hash_state,
+                                  const unsigned char *data,
+                                  int64_t nbytes) {
+#ifndef NUMTHREADS
+#define NUMTHREADS 8
+#endif
+  const uint64_t numthreads = NUMTHREADS;
+  uint64_t threadhash[numthreads+2];
+  //CHECK(numthreads > 0);
+  const uint64_t blocksz = 64; // cache block alignment (alternative: page size)
```

Comment:
it's going to be unbelievably verbose. The more verbose things are, the more difficult it is to read code. Using "sz" suffix is standard in the linux kernel. See here for example:
https://github.com/torvalds/linux/blob/5924bbecd0267d87c24110cbe2041b5075173a25/arch/microblaze/include/asm/mmu.h#L101